sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2360,7 +2360,8 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
       operationNotAllowed(
         s"Values should not be specified for key(s): ${badKeys.mkString("[", ",", "]")}", ctx)
     }
-    props.keys.toSeq
+    val cleanedTableProperties = cleanTableProperties(ctx, props)
```

Comment:
why we need to clean here but not `visitPropertyKeyValues`?

Reply:
visitPropertyKeyValues need extra information to filter in create/replace syntaxes. but here we just properties itself only this method is only used in ALTER TABLE UNSET only and properties remain map type here

Reply:
what if we call `visitPropertyKeys` in other places in the future? I think it's better to clean the properties in `visitAlterTableUnset`. e.g.
```
val keys = ...
val cleaned = cleanTableProperties(keys.map(_ -> "").toMap).keys.toSeq
```

Reply:
ok, I got this